### DISCUSSION TITLE
Help with SMF, how to set up localhost to reach out?

### DISCUSSION THREAD

**Discussion by melospoker** [community/community#57686] [Programming Help]
### Select Topic Area

Question

### Body

Please help me set up with either of the ngrok or localhost to reach out outside of my kali linux, thanks!

**Comment by Rimkomatic** [comment]
To set up localhost in Kali Linux, follow these steps:

1. Install Apache web server: 

   `sudo apt-get update`

   `sudo apt-get install apache2`

2. Start the Apache service:

   `sudo systemctl start apache2`

3. Verify that Apache is running:

   `sudo systemctl status apache2`

4. Create a sample HTML file to test the server:

   `sudo nano /var/www/html/index.html`

   Add some text to the file and save it.

5. Open your web browser and type `http://localhost` in the address bar. You should see the contents of the index.html file.

6. To reach out to the localhost from another machine, you need to find the IP address of your Kali Linux machine:

   `ip addr show`

   Look for the "inet" address under the interface you are using (e.g. eth0, wlan0).

7. On the other machine, open a web browser and type `http://` in the address bar. You should see the contents of the index.html file.

Note: If you have a firewall enabled on your Kali Linux machine, you may need to open port 80 (HTTP) to allow incoming connections.

**Reply by melospoker** [reply]
Thank you for such a great tutorial and detailed information!
I have done everything and can see that apache service its running on my Kali machine. When I type localhost inside of kali it shows the test that I created on the nano file.

The only hard time im having now its trying to rich it from a diff comp on another network. I found out the "inet address" but when I type it on a browser it shows nothing.

**Reply by Rimkomatic** [reply]
You're welcome

**Reply by melospoker** [reply]
https://prnt.sc/nm1_K7SvlmLl this is what I get but the problem I have is reaching that localhost:5555 outside kali.
